### PR TITLE
PBI Transaction History

### PR DESCRIPTION
**Pull request by catherineeangel**
In this PR, I have applied the transaction history feature for the app. I have applied this so that user can continue pending transaction that has not expired.

### CODE REVIEW

**Review by JohannesSetiawan — CHANGES_REQUESTED** [review]
Good implementation, but there's some issue detected on SonarCloud. Might want to fix that first

File: src/app/profile/TransactionHistory.tsx [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+                        </td>
+
+                        <td className="p-2 border-2 border-emerald-500">
+                          {name}
+                        </td>
+                        <td className="p-2 border-2 border-emerald-500">
+                          {formatRupiah(price)}
+                        </td>
+                        <td className="p-2 border-2 border-emerald-500">
+                          {dayjs(expiry_time).format('ddd, D MMM YY HH:mm')}
```
Comment (emiriko) [inline src/app/profile/TransactionHistory.tsx:84]: I think there are plenty of time in our projects that we use dayjs and format it to a certain format. I think it's better to make utils function for better code readability.

File: src/app/payment/PaymentDetail.tsx [hunk 2]
```diff
@@ -1,18 +1,34 @@
...
+
+  if (status == 'settlement' || status == 'capture')
+    return (
+      <PaymentSuccess
+        status={status ?? ''}
+        packageName={data?.transaction_detail.package.name ?? ''}
+      />
+    )
+  if (transactionErrorStatuses.includes(status ?? ''))
+    return <PaymentError status={status ?? ''} />
```
Comment (emiriko) [inline src/app/payment/PaymentDetail.tsx:31]: In here, is there really a case where status is null? If there are, I think you could define status as a type of possible values to reduce confusion

**Review by emiriko — CHANGES_REQUESTED** [review]
I really like your code, but I think there is a slight room for improvement. Thank you for your effort!

File: src/app/profile/TransactionHistory.tsx (hunk 1 above)
Comment (catherineeangel) [reply src/app/profile/TransactionHistory.tsx]: ok thanks for suggesting this, I have rafactored the code and make it more modular.

File: src/app/payment/PaymentDetail.tsx (hunk 2 above)
Comment (catherineeangel) [reply src/app/payment/PaymentDetail.tsx]: You're right. Thanks for pointing that outl.

**Review by JohannesSetiawan — APPROVED** [review]

**Review by emiriko — APPROVED** [review]
All good! Good job.

**Pull request merged**